### DISCUSSION TITLE
Difference in numeric calculations between .NET FW & 8

### DISCUSSION THREAD

**Discussion by karakasa** [dotnet/runtime#95050] [Q&A] [answered]
Recently I tried to port a [numeric calculation algorithm](https://gamma.cs.unc.edu/RVO2/downloads/cs-2.0/rvo2-cs-2.0.1.zip) from .NET FW to .NET 8 and found the result is very different. For the example `Circle` program, .NET FW takes `708.75` steps to coverage while .NET 8 takes `1024`. Both runs are in `Debug` mode.

The algorithm contains a considerable amount of code. Any idea on how to debug this?

**Comment by huoyaoyuan** [comment]
You can use `FloatToUInt32Bits` to compare the exact value of floats. The ToString formatting changed at .NET Core 3.
I don't expect the calculations to change though.

**Reply by karakasa** [reply]
You are right. The binary rep of floats stays the same.

Any idea on how to debug such issue, given that the algorithm is quiet complicated?

**Reply by huoyaoyuan** [reply, marked as answer]
I'd suggest printing logs and compare, as a first step to find the difference.

**Reply by Xyncgas** [reply]
Maybe implement GetHashCode which using System.Runtime.Interops.MemoryMarshal to convert float to Span`bytes and hash it using your favorite algorithm
